src/Hooks/useSettings.jsx

```diff
@@ -0,0 +1,33 @@
+import React, { createContext } from 'react';
+import PropTypes from 'prop-types';
+
+export const SettingsContext = createContext(null);
+
+export class SettingsProvider extends React.PureComponent {
+  constructor(props) {
+    super(props);
+    this.state = {
+      theme: {
+        isShowSettingsPage: false,
+        toggleSettingsPageHandler: this.toggleSettingsPage,
```

Comment:
`toggleSettings` is more than enough for name

Reply:
```suggestion
        toggleSettings: this.toggleSettings,
```
fixed